src/ip_notifier.cpp

```diff
@@ -348,10 +348,9 @@ struct ip_change_notifier_impl final : ip_change_notifier
 struct ip_change_notifier_impl final : ip_change_notifier
 {
 	explicit ip_change_notifier_impl(io_service& ios)
-		: m_hnd(ios, WSACreateEvent())
+		: m_ios(ios)
 	{
-		if (!m_hnd.is_open()) aux::throw_ex<system_error>(WSAGetLastError(), system_category());
-		m_ovl.hEvent = m_hnd.native_handle();
+		NotifyUnicastIpAddressChange(AF_UNSPEC, address_change_cb, this, false, &m_hnd);
```

Comment:
since this class now owns `m_hwnd`, it needs to either implement move constructor and move assignment (or delete them)

Reply:
Right. This class is non-movable now anyways due to the mutex, will delete them.

Reply:
Wait no, mutex is moveable. I'll implement the move functions.

Reply:
Haha no I had it right the first time, mutexes aren't movable so neither is this class.